tests/Integration/ApiPlatform/ModuleEndpointTest.php

```diff
@@ -220,6 +225,46 @@ public function testUpdateModuleStatus(array $module): void
         $this->assertEquals(0, $disabledModules['totalItems']);
     }
 
+    /**
+     * @depends testBulkUpdateStatus
+     */
+    public function testResetModule(array $module): void
+    {
+        // Reset specific module
+        $bearerToken = $this->getBearerToken(['module_read', 'module_write']);
+        $response = static::createClient()->request('PUT', sprintf('/module/%s/reset', $module['technicalName']), [
+            'auth_bearer' => $bearerToken,
+            'json' => [
+                'keepData' => false,
+            ],
+        ]);
+        self::assertResponseStatusCodeSame(200);
+        $decodedResponse = json_decode($response->getContent(), true);
+        $this->assertNotFalse($decodedResponse);
+
+        // Check response from status update request
+        $expectedModuleInfos = [
+            'moduleId' => $module['moduleId'],
+            'technicalName' => $module['technicalName'],
+            'version' => $module['version'],
+            'enabled' => true,
+            'installed' => true,
+        ];
+        $this->assertEquals($expectedModuleInfos, $decodedResponse);
+    }
+
+    public function testResetModule_NotInstalled(): void
+    {
+        $module = [
+            'technicalName' => 'ps_notthere',
```

Comment:
Actually, here you are testing the reset on a module that is not present at all This should return a 404 as I commented in the CQRS PR You can update this test once the CQRS command has been updated with the proper `ModuleNotFoundException`

Then you will have to add a mapping in the Api resource class (I added another comment explaining that), when it's done you can update this test to check that a 404 is correctly triggered when accessing non existent modules

You can even rename this one into `testModuleNotFound` and check that all the future endpoints will have this 404 behaviour (including the existing GET endpoint)